### DISCUSSION TITLE
Fabric profile not launching, the json file for fabric version is missing.

### DISCUSSION THREAD

**Discussion by MasterReady12** [FabricMC/fabric-api#1940] [Player support]
The JSON file for fabric version gets automatically deleted after launching the fabric profile from the launcher. The other jar file in the fabric version folder is of 0KB.
![Capture](https://user-images.githubusercontent.com/96872679/147753575-d1eee283-8878-4316-91c2-f15bb0fdc9c7.JPG)

The error message goes like this-
Unhandled exception has occurred in your application. If you click Continue, the application will ignore this error and attempt to continue. If you click Quit, the application will close immediately.

Directory 'fabric-loader-0.12.12-1.17.1' doesn't contain JSON file. Path: C:\Users\Admin\AppData\Roaming\.minecraft\versions\fabric-loader-0.12.12-1.17.1.

Details-

See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
dotMCLauncher.Versioning.VersionCorruptedOrNotExists: Directory 'fabric-loader-0.12.12-1.17.1' doesn't contain JSON file. Path: C:\Users\Admin\AppData\Roaming\.minecraft\versions\fabric-loader-0.12.12-1.17.1
   at dotMCLauncher.Versioning.VersionManifest.IsValid(DirectoryInfo pathToDirectory, Boolean throwsExceptions) in C:\Users\Jernej\source\repos\FreeLauncher-master\src\dotMCLauncher.Versioning\VersionManifest\VersionManifest.cs:line 177
   at dotMCLauncher.Versioning.VersionManifest.ParseVersion(DirectoryInfo pathToDirectory, Boolean parseInheritableVersion) in C:\Users\Jernej\source\repos\FreeLauncher-master\src\dotMCLauncher.Versioning\VersionManifest\VersionManifest.cs:line 155
   at FreeLauncher.Forms.LauncherForm.versionsListView_ItemMouseClick(Object sender, ListViewItemEventArgs e)
   at Telerik.WinControls.UI.RadListViewElement.OnItemMouseClick(ListViewItemEventArgs args)
   at Telerik.WinControls.UI.BaseListViewVisualItem.OnClick(EventArgs e)
   at Telerik.WinControls.RadElement.DoClick(EventArgs e)
   at Telerik.WinControls.UI.DetailListViewDataCellElement.OnClick(EventArgs e)
   at Telerik.WinControls.RadElement.DoClick(EventArgs e)
   at Telerik.WinControls.RadElement.RaiseBubbleEvent(RadElement sender, RoutedEventArgs args)
   at Telerik.WinControls.RadItem.RaiseBubbleEvent(RadElement sender, RoutedEventArgs args)
   at Telerik.WinControls.RadElement.RaiseRoutedEvent(RadElement sender, RoutedEventArgs args)
   at Telerik.WinControls.ComponentInputBehavior.OnMouseUp(MouseEventArgs e)
   at Telerik.WinControls.UI.RadListView.OnMouseUp(MouseEventArgs e)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at Telerik.WinControls.RadControl.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4110.0 built by: NET48REL1LAST_B
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
Launcher
    Assembly Version: 3.9.5.0
    Win32 Version: 3.9.5.0
    CodeBase: file:///C:/Users/Admin/AppData/Roaming/.minecraft/MCLauncher.exe
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4110.0 built by: NET48REL1LAST_B
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
Telerik.WinControls
    Assembly Version: 2019.1.117.40
    Win32 Version: 3.9.5.0
    CodeBase: file:///C:/Users/Admin/AppData/Roaming/.minecraft/MCLauncher.exe
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4110.0 built by: NET48REL1LAST_B
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3761.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
Telerik.WinControls.UI
    Assembly Version: 2019.1.117.40
    Win32 Version: 3.9.5.0
    CodeBase: file:///C:/Users/Admin/AppData/Roaming/.minecraft/MCLauncher.exe
----------------------------------------
CommandLine
    Assembly Version: 2.2.1.0
    Win32 Version: 3.9.5.0
    CodeBase: file:///C:/Users/Admin/AppData/Roaming/.minecraft/MCLauncher.exe
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4110.0 built by: NET48REL1LAST_B
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
Newtonsoft.Json
    Assembly Version: 12.0.0.0
    Win32 Version: 3.9.5.0
    CodeBase: file:///C:/Users/Admin/AppData/Roaming/.minecraft/MCLauncher.exe
----------------------------------------
System.Numerics
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3761.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Numerics/v4.0_4.0.0.0__b77a5c561934e089/System.Numerics.dll
----------------------------------------
System.Runtime.Serialization
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4110.0 built by: NET48REL1LAST_B
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.Serialization/v4.0_4.0.0.0__b77a5c561934e089/System.Runtime.Serialization.dll
----------------------------------------
System.Data
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3761.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_64/System.Data/v4.0_4.0.0.0__b77a5c561934e089/System.Data.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3761.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
fNbt
    Assembly Version: 0.6.4.0
    Win32 Version: 3.9.5.0
    CodeBase: file:///C:/Users/Admin/AppData/Roaming/.minecraft/MCLauncher.exe
----------------------------------------
TelerikCommon
    Assembly Version: 2019.1.117.40
    Win32 Version: 3.9.5.0
    CodeBase: file:///C:/Users/Admin/AppData/Roaming/.minecraft/MCLauncher.exe
----------------------------------------
Microsoft.VisualBasic
    Assembly Version: 10.0.0.0
    Win32 Version: 14.8.3761.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/Microsoft.VisualBasic/v4.0_10.0.0.0__b03f5f7f11d50a3a/Microsoft.VisualBasic.dll
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3761.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
Microsoft.GeneratedCode
    Assembly Version: 1.0.0.0
    Win32 Version: 4.8.3761.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
Telerik.WinControls.Themes.VisualStudio2012Dark
    Assembly Version: 2019.1.117.40
    Win32 Version: 3.9.5.0
    CodeBase: file:///C:/Users/Admin/AppData/Roaming/.minecraft/MCLauncher.exe
----------------------------------------
System.Management
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3761.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Management/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Management.dll
----------------------------------------
dotMCLauncher.Versioning
    Assembly Version: 18.11.25.57
    Win32 Version: 3.9.5.0
    CodeBase: file:///C:/Users/Admin/AppData/Roaming/.minecraft/MCLauncher.exe
----------------------------------------
Microsoft.mshtml
    Assembly Version: 7.0.3300.0
    Win32 Version: 7.0.3300.0
    CodeBase: file:///C:/Windows/assembly/GAC/Microsoft.mshtml/7.0.3300.0__b03f5f7f11d50a3a/Microsoft.mshtml.dll
----------------------------------------
dotMCLauncher.Profiling
    Assembly Version: 18.11.25.320
    Win32 Version: 3.9.5.0
    CodeBase: file:///C:/Users/Admin/AppData/Roaming/.minecraft/MCLauncher.exe
----------------------------------------
dotMCLauncher
    Assembly Version: 18.7.5.28
    Win32 Version: 3.9.5.0
    CodeBase: file:///C:/Users/Admin/AppData/Roaming/.minecraft/MCLauncher.exe
----------------------------------------
DotNetZip
    Assembly Version: 1.11.0.0
    Win32 Version: 3.9.5.0
    CodeBase: file:///C:/Users/Admin/AppData/Roaming/.minecraft/MCLauncher.exe
----------------------------------------
dotMCLauncher.Resourcing
    Assembly Version: 18.7.5.19
    Win32 Version: 3.9.5.0
    CodeBase: file:///C:/Users/Admin/AppData/Roaming/.minecraft/MCLauncher.exe
----------------------------------------
dotMCLauncher.Yggdrasil
    Assembly Version: 18.7.5.185
    Win32 Version: 3.9.5.0
    CodeBase: file:///C:/Users/Admin/AppData/Roaming/.minecraft/MCLauncher.exe
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.

**Comment by MasterReady12** [comment]
![Capture 2](https://user-images.githubusercontent.com/96872679/147753793-61eb21f5-c305-4e3a-bb6e-5c6d3c82d11c.JPG)
This is the error message. Also, I am using a hacked client.

**Comment by warjort** [comment]
Hacked clients are not supported, they are a breach of Minecraft's EULA
https://account.mojang.com/documents/minecraft_eula

Even if it was legit, you showing an error in some .net code.
Fabric and Minecraft are java programs.
